### PR TITLE
Add playlist FileItem classify

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Move playlist fileitem classifiers to separate file

## Motivation and context
Reduce CFileItem interface

## How has this been tested?
Builds + I have added tests

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/music/windows/GUIWindowMusicBase.cpp [hunk 1]
```diff
@@ -517,7 +517,10 @@ bool CGUIWindowMusicBase::OnContextButton(int itemNumber, CONTEXT_BUTTON button)
 
   case CONTEXT_BUTTON_EDIT_SMART_PLAYLIST:
     {
-      std::string playlist = item->IsSmartPlayList() ? item->GetPath() : m_vecItems->GetPath(); // save path as activatewindow will destroy our items
+      std::string playlist =
```
Comment (CrystalP) [inline xbmc/music/windows/GUIWindowMusicBase.cpp:520]: might as well throw in a const.

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 2]
```diff
@@ -940,7 +940,10 @@ bool CGUIWindowVideoBase::OnContextButton(int itemNumber, CONTEXT_BUTTON button)
     return true;
   case CONTEXT_BUTTON_EDIT_SMART_PLAYLIST:
     {
-      std::string playlist = m_vecItems->Get(itemNumber)->IsSmartPlayList() ? m_vecItems->Get(itemNumber)->GetPath() : m_vecItems->GetPath(); // save path as activatewindow will destroy our items
+      std::string playlist =
```
Comment (CrystalP) [inline xbmc/video/windows/GUIWindowVideoBase.cpp:943]: const?

File: xbmc/playlists/PlayListFileItemClassify.cpp [hunk 3]
```diff
@@ -19,4 +21,13 @@ bool IsPlayList(const CFileItem& item)
   return CPlayListFactory::IsPlaylist(item);
 }
 
+bool IsSmartPlayList(const CFileItem& item)
+{
+  if (item.HasProperty("library.smartplaylist") &&
+      item.GetProperty("library.smartplaylist").asBoolean())
```
Comment (CrystalP) [inline xbmc/playlists/PlayListFileItemClassify.cpp:27]: `item.GetProperty("library.smartplaylist").asBoolean(false)`?

**Comment by CrystalP** [comment]
I probably didn't noticed in previous PR, the copyright years of the new files seem strange? I don't know the rules though, is that on purpose?
`Copyright (C) 2005-2018 Team Kodi` for TestPlayListFileItemClassify.cpp, PlayListFileItemClassify.h, PlayListFileItemClassify.cpp

File: xbmc/music/windows/GUIWindowMusicBase.cpp (hunk 1 above)
Comment (notspiff) [reply xbmc/music/windows/GUIWindowMusicBase.cpp]: ok

File: xbmc/video/windows/GUIWindowVideoBase.cpp (hunk 2 above)
Comment (notspiff) [reply xbmc/video/windows/GUIWindowVideoBase.cpp]: ok

File: xbmc/playlists/PlayListFileItemClassify.cpp (hunk 3 above)
Comment (notspiff) [reply xbmc/playlists/PlayListFileItemClassify.cpp]: ok

**Comment by notspiff** [comment]
no real though given to copyrights. bumped them to 2024.

**Pull request merged**